src/nv/storage/index.rs

```diff
@@ -105,3 +126,286 @@ impl From<NvIndexAttributes> for TPMA_NV {
+        self.nv_index_attributes.set_policy_read(set);
+        self
+    }
+
+    /// Controls the `no DA` attribute
+    ///
+    /// # Arguments
+    /// * `set` - `true` indicates that the attribute should have the value SET.
+    ///           `false`indicates that the attribute should have the value CLEAR.
+    pub fn with_no_da(mut self, set: bool) -> Self {
+        self.nv_index_attributes.set_no_da(set);
+        self
+    }
+
+    /// Controls the `orderly` attribute
+    ///
+    /// # Arguments
+    /// * `set` - `true` indicates that the attribute should have the value SET.
+    ///           `false`indicates that the attribute should have the value CLEAR.
+    pub fn with_orderly(mut self, set: bool) -> Self {
+        self.nv_index_attributes.set_orderly(set);
+        self
+    }
+
+    /// Controls the `clear stclear` attribute
+    ///
+    /// # Arguments
+    /// * `set` - `true` indicates that the attribute should have the value SET.
+    ///           `false`indicates that the attribute should have the value CLEAR.
+    pub fn with_clear_stclear(mut self, set: bool) -> Self {
+        self.nv_index_attributes.set_clear_stclear(set);
+        self
+    }
+
+    /// Controls the `read locked` attribute
+    ///
+    /// # Arguments
+    /// * `set` - `true` indicates that the attribute should have the value SET.
+    ///           `false`indicates that the attribute should have the value CLEAR.
+    pub fn with_read_locked(mut self, set: bool) -> Self {
+        self.nv_index_attributes.set_read_locked(set);
+        self
+    }
+
+    /// Controls the `written` attribute
+    ///
+    /// # Arguments
+    /// * `set` - `true` indicates that the attribute should have the value SET.
+    ///           `false`indicates that the attribute should have the value CLEAR.
+    pub fn with_written(mut self, set: bool) -> Self {
+        self.nv_index_attributes.set_written(set);
+        self
+    }
+
+    /// Controls the `platform create` attribute
+    ///
+    /// # Arguments
+    /// * `set` - `true` indicates that the attribute should have the value SET.
+    ///           `false`indicates that the attribute should have the value CLEAR.
+    pub fn with_platform_create(mut self, set: bool) -> Self {
+        self.nv_index_attributes.set_platform_create(set);
+        self
+    }
+
+    /// Controls the `read stclear` attribute
+    ///
+    /// # Arguments
+    /// * `set` - `true` indicates that the attribute should have the value SET.
+    ///           `false`indicates that the attribute should have the value CLEAR.
+    pub fn with_read_stclear(mut self, set: bool) -> Self {
+        self.nv_index_attributes.set_read_stclear(set);
+        self
+    }
+
+    /// Builds the nv index attributes.
+    ///
+    /// # Errors
+    /// Returns an error if some attributes are missing
+    /// or are in conflict with eachother.
+    pub fn build(self) -> Result<NvIndexAttributes> {
```

Comment:
Would it be useful to also provide this as a method on `NvIndexAttributes`, something like `fn check(&self) -> Result<()>`

Reply:
Maybe, I have not really thought about it. What I have been thinking about is to be able to initialize the builder using attributes. That would also give a way to verify that they are ok by then initializing a builder with the attributes you want to check and then call build on it. Though not so intuitive and user friendly perhaps.

```NvIndexAttributesBuilder::create_from_attributes(nv_index_attributes)```

Reply:
Hmm, I think it's ok as it is now, but maybe it's a good idea to change the `From<TPMA_NV>` above to a `TryFrom` that also does the checks? Don't know if by using `create_from_attributes` or in some other way, up to you!

Reply:
I did all of it